src/Parsers/ParserCreateFunctionQuery.cpp

```diff
@@ -167,6 +169,20 @@ bool ParserCreateFunctionQuery::parseImpl(IParser::Pos & pos, ASTPtr & node, Exp
                 throw Exception("AUTH_METHOD must be 'none' or 'auth_header'", ErrorCodes::UNKNOWN_FUNCTION);
             }
         }
+        else
```

Comment:
remove the 'else'

Reply:
This can ensure that the first one of url's children is always AUTH_METHOD.